### PR TITLE
Fix warning for nonexistent inventory cache

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes a confusing warning when an inventory cache file doesn't exist.

cc @briantist

Alternative to #57646

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Inventory cache

### CODE REVIEW

**Comment by briantist** [comment]
thanks for this @s-hertel !

File: lib/ansible/plugins/cache/__init__.py [hunk 1]
```diff
@@ -320,10 +320,7 @@ def __len__(self):
     def _do_load_key(self, key):
         load = False
         if key not in self._cache and key not in self._retrieved and self._plugin_name != 'memory':
-            if isinstance(self._plugin, BaseFileCacheModule):
-                load = True
-            elif not isinstance(self._plugin, BaseFileCacheModule) and self._plugin.contains(key):
-                # Database-backed caches don't raise KeyError for expired keys, so only load if the key is valid by checking contains()
+            if self._plugin.contains(key):
```
Comment (bcoca) [inline lib/ansible/plugins/cache/__init__.py]: at this point, a single `if` is needed (maby use ~'any'~ 'all' to make it more readable?)

**Review by bcoca — APPROVED** [review]

**Comment by s-hertel** [comment]
I'm adding a test too so this can be backported.

**Pull request merged**